File: Homework 4/C++/Address.cpp
//#include <locale>
#include "Address.h"
#include <ctype.h>

using namespace std;

Address::Address()
{
	street = string();
	city = SingleName();
	state = SingleName();
	int zip = 0;
}

Address::~Address()
{
}

void Address::setStreet(string input) {
	street = input;
}

string Address::getStreet() {
	return street;
}

void Address::setCity(SingleName input) {
	city = input;
}

SingleName Address::getCity() {
	return city;
}

void Address::setState(SingleName input) {
	state = input;
}

SingleName Address::getState() {
	return state;
}

void Address::setZip(int input) {
	zip = input;
}

int Address::getZip() {
	return zip;
}

bool Address::checkZip(string input) {
	int i;
	for (i = 0; i < input.length(); i++) {
		if (!isdigit(input[i]))
			return false;
	}
	return true;
}

File: Homework 4/JAVA/Address.java
/*
 * Objective: Creates an address object that includes street, city, state and zip
*/

public class Address {
	
	private String street;
	private SingleName city, state;
	private int zip;
	
public Address(String street, SingleName city, SingleName state, int zip) {
		super();
		this.street = street;
		this.city = city;
		this.state = state;
		this.zip = zip;
	}

public Address() {
	
}

void setStreet(String street ) {
	this.street = street;
}

void setCity(SingleName city) {
	this.city = city;
}

void setState(SingleName state) {
	this.state = state;
}

void setZip(int zip) {
	this.zip = zip;
}

//first line with street number and name
//listed in instructions as "getfFrst"
String getStreet() {

	return street;
}

SingleName getCity() {
	
	return city;
}

SingleName getState() {
	
	return state;
}

int getZip() {
	
	return zip;
}
}
